server/odc-service/src/main/java/com/oceanbase/odc/service/permission/DBResourcePermissionHelper.java

```diff
@@ -425,7 +425,7 @@ private Map<Long, Set<DatabasePermissionType>> getInnerDBPermissionTypes(Collect
                 .findNotExpiredByUserIdAndDatabaseIdIn(authenticationFacade.currentUserId(), databaseIds)
                 .stream().collect(Collectors.toMap(
                         UserTablePermissionEntity::getDatabaseId,
-                        e -> Collections.singleton(DatabasePermissionType.ACCESS),
+                        e -> Collections.singleton(DatabasePermissionType.from(e.getAction())),
```

Comment:
group by will be better, cause that there will be multi tables that have the same database id

Reply:
done~